_SPIKE_.md

```diff
@@ -87,6 +87,8 @@ class Product extends Entity {
 interface ProductLinks {
   category?: Category & CategoryLinks;
```

Comment:
Can we use our shiny new type here like `category?: CategoryWithLinks`? (same for `products?:` inside `CategoryLinks`)